source/backend/api/Services/FormDocumentService.cs

```diff
@@ -0,0 +1,136 @@
+            _formTypeRepository = formTypeRepository;
+            _documentService = documentService;
+            _documentRepository = documentRepository;
+            _acquisitionFileFormRepository = acquisitionFileFormRepository;
+        }
+
+        public IList<PimsFormType> GetAllFormDocumentTypes()
+        {
+            _logger.LogInformation("Getting all form document types");
+            this.User.ThrowIfNotAuthorized(Permissions.FormView);
+
+            var formTemplates = _formTypeRepository.GetAllFormTypes();
+            return formTemplates;
+        }
+
+        public IList<PimsFormType> GetFormDocumentTypes(string formTypeCode)
+        {
+            this.Logger.LogInformation("Getting document types for code {formTypeCode}", formTypeCode);
+            this.User.ThrowIfNotAuthorized(Permissions.FormView);
+
+            return new List<PimsFormType>() { _formTypeRepository.GetByFormTypeCode(formTypeCode) };
+        }
+
+        public async Task<DocumentUploadRelationshipResponse> UploadFormDocumentTemplateAsync(string formTypeCode, DocumentUploadRequest uploadRequest)
+        {
+            this.Logger.LogInformation("Uploading template for document form type");
+            this.User.ThrowIfNotAuthorized(Permissions.DocumentAdmin);
+
+            var currentFormType = _formTypeRepository.GetByFormTypeCode(formTypeCode);
+            if (currentFormType.DocumentId != null)
+            {
+                var result = await DeleteFormDocumentTemplateAsync(currentFormType);
+                if (result.Status != ExternalResultStatus.Success)
+                {
+                    throw new InvalidOperationException("Could not remove existing template");
+                }
+                else
+                {
+                    // Retrieve the form type again to get the latest after deletion.
+                    currentFormType = _formTypeRepository.GetByFormTypeCode(formTypeCode);
+                }
+            }
+
+            DocumentUploadResponse uploadResult = await _documentService.UploadDocumentAsync(uploadRequest);
+
+            DocumentUploadRelationshipResponse relationshipResponse = new DocumentUploadRelationshipResponse()
+            {
+                UploadResponse = uploadResult,
+            };
+
+            if (uploadResult.DocumentExternalResult.Status == ExternalResultStatus.Success && uploadResult.Document != null && uploadResult.Document.Id != 0)
+            {
+                currentFormType.DocumentId = uploadResult.Document.Id;
+                var updatedFormType = _formTypeRepository.SetFormTypeDocument(currentFormType);
+                _formTypeRepository.CommitTransaction();
+
+                relationshipResponse.DocumentRelationship = _mapper.Map<DocumentRelationshipModel>(updatedFormType);
+            }
+
+            return relationshipResponse;
+        }
+
+        public async Task<ExternalResult<string>> DeleteFormDocumentTemplateAsync(PimsFormType formType)
```

Comment:
This is an interesting case, we are only ever updating some metadata on this type rather then changing it.